src/main/java/org/opensearch/timeseries/transport/handler/SearchHandler.java

```diff
@@ -35,25 +36,32 @@ public class SearchHandler {
     private final Logger logger = LogManager.getLogger(SearchHandler.class);
     private final Client client;
     private volatile Boolean filterEnabled;
+    private final boolean shouldUseResourceAuthz;
 
     public SearchHandler(Settings settings, ClusterService clusterService, Client client, Setting<Boolean> filterByBackendRoleSetting) {
         this.client = client;
         filterEnabled = filterByBackendRoleSetting.get(settings);
+        shouldUseResourceAuthz = ParseUtils.shouldUseResourceAuthz();
         clusterService.getClusterSettings().addSettingsUpdateConsumer(filterByBackendRoleSetting, it -> filterEnabled = it);
     }
 
     /**
      * Validate user role, add backend role filter if filter enabled
      * and execute search.
      *
-     * @param request search request
+     * @param request        search request
+     * @param pair          paid of index name and id field
      * @param actionListener action listerner
      */
-    public void search(SearchRequest request, ActionListener<SearchResponse> actionListener) {
+    public void search(SearchRequest request, Pair<String, String> pair, ActionListener<SearchResponse> actionListener) {
```

Comment:
please review this. This change currently modifies all `searchHandler.search` calls to pass the index name and id field of the config. However, the search calls that are made from tasks and results indices are not covered by resource-authz. Only AD and forecast config index are, since those two are registered as protected indices.